events/api/src/main/java/org/projectnessie/events/api/Event.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.immutables.value.Value;
+import org.projectnessie.events.api.json.EventTypeIdResolver;
+
+/**
+ * Base interface for all events produced by Nessie.
+ *
+ * @see CommitEvent
+ * @see MergeEvent
+ * @see TransplantEvent
+ * @see ReferenceCreatedEvent
+ * @see ReferenceUpdatedEvent
+ * @see ReferenceDeletedEvent
+ * @see ContentStoredEvent
+ * @see ContentRemovedEvent
+ */
+@JsonSubTypes({
+  @JsonSubTypes.Type(name = "COMMIT", value = ImmutableCommitEvent.class),
```

Comment:
Nit: remove `Immutable` from the `value` attributes.

Reply:
Actually, it seems that `@JsonSubTypes` is not needed anymore: we are using a custom strategy instead to resolve the type ids.